File: src/main/java/com/example/chatService/repository/MessageRepository.java
package com.example.chatService.repository;

import com.example.chatService.entity.Message;
import com.example.chatService.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface MessageRepository extends JpaRepository<Message, String> {
    List<Message> findBySenderAndReceiverOrderBySentAtAsc(User sender, User receiver);
    List<Message> findByReceiverAndIsReadFalse(User receiver);
}

File: src/main/java/com/example/chatService/repository/MessageRepository.kt
package com.example.chatService.repository

import com.example.chatService.entity.Message
import com.example.chatService.entity.User
import org.springframework.data.jpa.repository.JpaRepository

interface MessageRepository : JpaRepository<Message, String> {
    fun findBySenderAndReceiverOrderBySentAtAsc(sender: User, receiver: User): List<Message>

    fun findByReceiverAndIsReadFalse(receiver: User): List<Message>
}
